progenitor-impl/src/cli.rs

```diff
@@ -477,12 +477,21 @@ impl Generator {
             let body_type_ident = body_type.ident();
             quote! {
                 if let Some(value) =
-                    matches.get_one::<std::path::PathBuf>("json-body")
+                    matches.get_one::<String>("json-body")
                 {
-                    let body_txt = std::fs::read_to_string(value).unwrap();
+                    use std::io::Read;
+                    let body_input = match value.as_str() {
+                        "-" => {
+                            let mut buf = Vec::new();
+                            std::io::stdin().read_to_end(&mut buf).unwrap();
+                            buf
+                        }
+                        file => std::fs::read(&file).unwrap()
+                    };
```

Comment:
Would `from_reader` simplify this at all? https://docs.rs/serde_json/latest/serde_json/fn.from_reader.html

Reply:
Definitely, updated it to use `from_reader` in both branches. Before merging though I would like to see if I can get a test in here. I don't think anything currently tests if a generated CLI would actually compile.

Reply:
That would be great